OpenVPNHelper/OpenVPNHelper.m

```diff
@@ -85,6 +85,43 @@ - (BOOL)verify:(NSString *)identifier atURL:(NSURL *)fileURL {
 }
 
 - (void)startOpenVPNAtURL:(NSURL *_Nonnull)launchURL withConfig:(NSURL *_Nonnull)config upScript:(NSURL *_Nullable)upScript downScript:(NSURL *_Nullable)downScript leasewatchPlist:(NSURL *_Nullable)leasewatchPlist leasewatchScript:(NSURL *_Nullable)leasewatchScript scriptOptions:(NSArray <NSString *>*_Nullable)scriptOptions reply:(void(^_Nonnull)(BOOL))reply {
+    
+    // Get the path of config file
+    NSString* path = config.path;
+    NSString* content = [NSString stringWithContentsOfFile:path
+                                                  encoding:NSUTF8StringEncoding
+                                                     error:NULL];
+    
+    NSMutableArray *listItems = [content componentsSeparatedByString:@"\r\n"];
+    NSArray *maliciousCommands = @[@"up", @"tls-verify", @"ipchange", @"client-connect", @"route-up",@"route-pre-down",@"client-disconnect",@"down",@"learn-address",@"auth-user-pass-verify"];
+    
+    
+    //loop through array to check if malicious is command
+    for ( int i = 0; i < [listItems count]; i++) {
+        NSString *line = [listItems objectAtIndex: i];
+        
+        
+        //Loop through malicious commands to check if any of them is present in the line
+        for ( int j = 0; j < [maliciousCommands count]; j++) {
+            NSString *maliciousCommand = [maliciousCommands objectAtIndex: j];
+            
+             //Check if malicious command was found
+            if ([line rangeOfString:[NSString stringWithFormat:@"%@%@", maliciousCommand,@" "]].location == NSNotFound) {
```

Comment:
Simpler: `[NSString stringWithFormat:@"%@%@", maliciousCommand,@" "]]` -> `[maliciousCommand stringByAppendingString:@" "]`

Reply:
There is another issue with this parsing. It would also match commented out lines, or lines where the text appears somewhere inside the options for a command where it could be valid. I think a better approach would be to trim spacing characters from the line, take the first n (where n is the length of the command + a space) characters of the line, and match those case insensitive to the disallowed commands.

Reply:
Ok, Going deep more

Reply:
This now also matches if for example it anywhere matches the phrase “up “.  It seems to me that this could happen soo this could result in false positives.  Use my suggestion of trimming spaces so that you can look if the comment appears at the beginning of the sentence.